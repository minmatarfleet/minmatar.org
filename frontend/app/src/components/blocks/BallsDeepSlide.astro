---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import type { CorporationObject } from '@dtypes/layout_components'

interface Props {
    corporation:            CorporationObject;
    is_user_corporation?:   boolean;
    picture_placement?:     'left' | 'right';
    
}

const {
    corporation,
    is_user_corporation = false,
    picture_placement = 'left',
} = Astro.props

const TWAN_MOLENAAR_ID = 2114993571

import CorporationSlide from '@components/blocks/CorporationSlide.astro';
---

<CorporationSlide
    id="ballsdeep-slide"
    ceo_image='/images/corporations/ballsdeep/ceo.png'
    ceo_character_id={TWAN_MOLENAAR_ID}
    corporation={corporation}
    is_user_corporation={is_user_corporation}
    picture_placement={picture_placement}
>
    <img slot="ships" class="[ vedmak ]" src="/images/corporations/ballsdeep/vedmak.png" />
    <img slot="ships" class="[ tengu ]" src="/images/corporations/ballsdeep/tengu.png" />
    <img slot="ships" class="[ orthrus ]" src="/images/corporations/ballsdeep/orthrus.png" />

    <p>{corporation.introduction ? corporation.introduction : t('balls_deep_expeditionary_description_fallback') }</p>
</CorporationSlide>

<style lang="scss">
    img {
        image-rendering: pixelated;
    }

    .vedmak {
        position: absolute;
        top: 40%;
        left: 5%;
        height: auto;
        width: 27%;
        z-index: 1;
        transform: scale(0);
    }

    .tengu {
        position: absolute;
        top: 35%;
        left: -40%;
        width: 65%;
        height: auto;
        z-index: 2;
        transform: scale(0);
    }

    .orthrus {
        position: absolute;
        left: -38%;
        top: 39%;
        width: 61%;
        height: auto;
        z-index: 1;
        transform: scale(0);
    }

    @keyframes vedmak {
        from {
            transform: scale(0);
            top: 40%;
            left: 5%;
        }

        to {
            transform: scale(1);
            top: 0%;
            left: 36%;
        }
    }

    @keyframes tengu {
        from {
            transform: scale(0);
            top: 35%;
            left: -40%;
        }

        to {
            transform: scale(1);
            top: 30%;
            left: 26%;
        }
    }

    @keyframes orthrus {
        from {
            transform: scale(0);
            left: -38%;
            top: 39%;
        }

        to {
            transform: scale(1);
            top: 49%;
            left: 23%;
        }
    }
</style>

<style lang="scss" is:global>
    #ballsdeep-slide .ceo {
        left: 7%;
    }

    .swiper-slide-active {
        .vedmak {
            animation: vedmak cubic-bezier(0.390, 0.575, 0.565, 1.000) both 0.45s;
            animation-delay: 2s;
            animation-iteration-count: 1;
            transform-origin: top right;
            animation-fill-mode: forwards;
        }
        
        .tengu {
            animation: tengu cubic-bezier(0.390, 0.575, 0.565, 1.000) both 0.45s;
            animation-delay: 2.2s;
            animation-iteration-count: 1;
            transform-origin: top right;
            animation-fill-mode: forwards;
        }
        
        .orthrus {
            animation: orthrus cubic-bezier(0.390, 0.575, 0.565, 1.000) both 0.45s;
            animation-delay: 2.4s;
            animation-iteration-count: 1;
            transform-origin: top right;
            animation-fill-mode: forwards;
        }
    }
</style>