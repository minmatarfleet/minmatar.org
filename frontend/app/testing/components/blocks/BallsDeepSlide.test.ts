import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import BallsDeepSlide from "@components/blocks/BallsDeepSlide.astro";

test("BallsDeepSlide defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(BallsDeepSlide, {
    props: {
      corporation: 5,
      is_user_corporation: true,
    },
  });

  expect(result).toMatchSnapshot();
});
