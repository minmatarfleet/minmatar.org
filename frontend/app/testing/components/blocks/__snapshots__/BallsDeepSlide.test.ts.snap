// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BallsDeepSlide defaults 1`] = `
"<div class="fixed-fluid fluid_first corporation-intro w-full items-center" id="ballsdeep-slide" data-astro-cid-c2ktugqm="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 650px;--fixed_fluid_breakpoint: 30%;">  <div class="wrapper" data-astro-cid-c2ktugqm="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-s);--wrapper_padding_inline_end: var(--space-s);--wrapper_padding_block: var(--space-s);--wrapper_max_width: 80rem;">  <div class="flex-block [ content ][ pb-32 lg:pb-0 ] gap---space-xl" data-astro-cid-c2ktugqm="true" data-mod style="--flex_block_gap: var(--space-xl);">  <div class="!flex !items-center" data-astro-cid-c2ktugqm="true">  <div class="fixed-fluid [ items-center w-full ]" data-astro-cid-c2ktugqm="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture class="[ border ]" data-astro-cid-c2ktugqm> <img loading="lazy" src="https://images.evetech.net/corporations/undefined/logo?size=64" width="64" height="64" alt="undefined corporation logo" data-astro-cid-c2ktugqm> </picture> <h2 data-astro-cid-c2ktugqm></h2>  </div>    </div> <div class="wrapper" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: 0;--wrapper_padding_inline_end: 0;--wrapper_padding_block: 0;--wrapper_max_width: var(--max-text-width);">        <p data-astro-cid-hpttrhaq>You dont know what will happen, we dont know what will happen. All we know is it will be Balls Deep. We kill things, in specialized ways, often outnumbered. Experienced and self sustaining players feel free to apply to kill things with us.</p>    </div>   <div class="flex-block gap---space-xl" id="corporation-status-undefined" x-data="{
        init() {
            
        }
    }" data-astro-cid-ggpwwpcg="true" data-mod style="--flex_block_gap: var(--space-xl);">  <div class="flex-inline" data-astro-cid-ggpwwpcg="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: flex-start;"> <button x-data="{
            show_corporation_modal() {
                    show_modal({
                        partial: '/partials/corporation_modal_component/?corporation_id=undefined',
                    })
                }
            }" x-on:click.stop.prevent="show_corporation_modal()" data-astro-cid-ggpwwpcg="true" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" disabled x-init="$el.removeAttribute(&#34;disabled&#34;)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   Learn more  </div>   </button>   </div>   <small class="[ status-text ]" data-astro-cid-ggpwwpcg>You are member of this corporation</small> </div>    </div>   </div>   <div class="[ corporation-identity ]" data-astro-cid-c2ktugqm> <picture data-astro-cid-c2ktugqm> <button class="[ ceo ]" x-data="{
                    show_character_modal() {
                        show_modal({
                            partial: '/partials/character_modal_component/?character_id=92000736',
                        })
                    }
                }" x-on:click="show_character_modal()" data-astro-cid-c2ktugqm="true" data-astro-cid-syldxee3="true">  <img src="/images/corporations/ballsdeep/ceo.png" height="496" width="940" data-astro-cid-c2ktugqm>  </button>  <button class="[ co-ceo ]" slot="coceo" x-data="{
            show_character_modal() {
                show_modal({
                    partial: '/partials/character_modal_component/?character_id=2114993571',
                })
            }
        }" x-on:click="show_character_modal()" data-astro-cid-hpttrhaq="true" data-astro-cid-syldxee3="true">  <img src="/images/corporations/ballsdeep/co-ceo.png" height="496" width="940" data-astro-cid-hpttrhaq>  </button>  <img class="[ ikitursa ]" src="/images/corporations/ballsdeep/ikitursa.png" data-astro-cid-hpttrhaq><img class="[ tengu ]" src="/images/corporations/ballsdeep/tengu.png" data-astro-cid-hpttrhaq><img class="[ orthrus ]" src="/images/corporations/ballsdeep/orthrus.png" data-astro-cid-hpttrhaq> </picture> </div>  </div>      "
`;
